src/properties.rs

```diff
@@ -74,15 +97,120 @@ pub enum Property<'a> {
     SharedSubscriptionAvailable(u8),
 }
 
-impl From<u32> for PropertyIdentifier {
-    fn from(val: u32) -> Self {
-        for entry in Self::into_enum_iter() {
-            if entry as u32 == val {
-                return entry;
+struct PropertyVisitor<'a> {
+    _data: core::marker::PhantomData<&'a ()>,
+}
+
+impl<'a, 'de: 'a> serde::de::Visitor<'de> for PropertyVisitor<'a> {
+    type Value = Property<'a>;
+
+    fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result {
+        write!(formatter, "enum Property")
+    }
+
+    fn visit_enum<A: serde::de::EnumAccess<'de>>(self, data: A) -> Result<Self::Value, A::Error> {
+        use serde::de::{Error, VariantAccess};
+
+        let (field, variant) = data.variant::<PropertyIdentifier>()?;
+        crate::trace!("Deserializing {:?}", field);
+
+        match field {
+            PropertyIdentifier::ResponseTopic => {
+                Ok(Property::ResponseTopic(variant.newtype_variant()?))
```

Comment:
This process should likely be offloaded into some type of macro.

Reply:
I think we could develop a proc macro similar to https://github.com/dtolnay/serde-repr to assign a property identifier to each `Property`.

Reply:
IMO those two big matches don't look too bad. I don't know whether a macro would help a lot.

Reply:
Maybe the `Ok()` can be factored out (the default arm can be `_ => return Err(...)`).